File: ws/ayuquina/u2/ws21zooWithInheritence/Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Bird.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Danny Ayuquina, LogicLegion, DCCO-ESPE
 */
public abstract class Bird extends Animal {

    public Bird(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }

}

File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Bird.js
const Animal = require('./Animal');

class Bird extends Animal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
        if (new.target === Bird) {
            throw new Error('Cannot instantiate abstract class Bird');
        }
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
}

module.exports = Bird;